### PR TITLE
Refactor of orb

### PR DESCRIPTION
**Pull request by dylanhitt**
An orb should be self contained and should require little to no effort for customers to use. 

The following changes the orb and ci code to better align with the circle ecosystem. These change will cause a major version bump. Also the java source should be removed from this repo. Something called like `nowsecure-auto`, this repo would be responsible for building the java .jar as well as a dockerfile with this tool bundled into it.

Changes:
- build dockerfile with the `nowsecure-auto` tool. Again this should removed
- removed unused files and .jars
- removes the need for gradle when running ci
- update CI to test, build, and publish the orb

- [ ] change all references from dylanhitt to nowsecure

### CODE REVIEW

File: src/@orb.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      token:
+        description: |
+          API token, visit
+          https://docs.nowsecure.com/auto/integration-services/jenkins-integration
+          to generate token
+        type: string
+        default: $AUTO_TOKEN
+
+      group:
+        description: Specify group if you belong to multiple groups
```
Comment (haeser) [inline src/@orb.yml:27]: This is the group id not the group name. Right?
```suggestion
      group:
        description: Specify `group_id` if you belong to multiple groups
```

perhaps even a more drastic rename?

```suggestion
      group_id:
        description: Specify group uuid if you belong to multiple groups. You can obtain the group id by hovering over the group name in the [Account & Management > Groups](https://lab.nowsecure.com/account/management/groups)
```

**Review by haeser — COMMENTED** [review]
Very, very nice 👏🏼
Added a few inline comments/questions.

An example is probably already in the making? If not adding one example would be super helpful. Especially when browsing through the [orb registry](https://circleci.com/developer/orbs) to quickly see how easy it will be to integrate that orb.

Perhaps something along the lines?
`src/examples/example.yml`

``` yaml
description: >
  
usage:
  version: 2.1
  orbs:
    auto_ci: dylanhitt/ci-auto-orb@2.3.4
  workflows:
    nowsecure-scan:
      jobs:
        - auto_ci/mobile_security_test:
             file: "app.apk"
             score: 50
             wait: 60
             show_status_messages: true
             context: 
               nowsecure-secrets
```

File: src/@orb.yml (hunk 1 above)
Comment (dylanhitt) [reply src/@orb.yml:27]: Nice catch

**Comment by dylanhitt** [comment]
@haeser thank you for looking at this. It's bit of a WIP as I'd like to move the java code out of it before we move forward anymore. But yes examples to come 😄

File: src/@orb.yml (hunk 1 above)
Comment (dylanhitt) [reply src/@orb.yml:27]: I think changing the java source would better reflect this.

File: src/@orb.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      - run:
+          command: >
+            nowsecure-auto --url << parameters.url >> \
+              --token << parameters.token >> \
+              --artifact-dir << parameters.artifact_dir >> \
+              --file << parameters.file >> \
+              --group << parameters.group >> \
+              --wait << parameters.wait >> \
+              --score << parameters.score >> \
+              --show-status-messages << parameters.show_status_messages >>
```
[the remark this answers is not in the document]
Comment (dylanhitt) [reply src/@orb.yml:96]: @haeser it turns out this wasn't needed. Pleas resolve when you have a moment.

Comment (haeser) [reply src/@orb.yml:96]: couldn't resolve it (perhaps missing a permission) but I was able to delete the original suggestion

**Pull request closed without merging**